src/gt4py/next/type_system/type_specifications.py

```diff
@@ -92,31 +79,37 @@ def __str__(self) -> str:
         return f"{kind_str}{self.shape}"
 
 
-@dataclass(frozen=True)
+class ListType(DataType):
+    element_type: DataType
+    # TODO(havogt): the `offset_type` is not yet used in type_inference,
+    # it is meant to describe the neighborhood (via the local dimension)
+    offset_type: Optional[common.Dimension] = None
+
+
+class FieldType(DataType, CallableType):
+    dims: list[common.Dimension]
+    dtype: ScalarType | ListType
```

Comment:
Unrelated to this PR which just exposes this, but this looks inconsistent. Our conclusion with respect to list types was that the frontend has the local dimension in `dims` while ITIR uses `ListType`. Consequently frontend and ITIR have different concepts of what a Field is should therefore also use different `FieldType`s.

Reply:
I would propose to document this here, because ultimately we should have one FieldType so refactoring in two different is probably quite some work that will disappear later.

Reply:
Makes sense.